xarray/core/dataset.py

```diff
@@ -2494,12 +2501,13 @@ def update(self, other, inplace=True):
             If any dimensions would have inconsistent sizes in the updated
             dataset.
         """
+        # _check_inplace(inplace)
```

Comment:
@shoyer What's the right thing to do here? Changing the default to None/False breaks a lot of test.

Reply:
`Dataset.update()` has always been an in-place operation for consistency with `dict.update()`. I don't think it really makes sense to change that.

I would suggest that we do the deprecation the other direction here: remove the `inplace` argument but keep the behavior as always inplace.